cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -679,6 +682,7 @@ private void startExecution(LoadMonitor loadMonitor,
                               boolean isTriggeredByUserRequest) throws OngoingExecutionException {
     _executionStoppedByUser.set(false);
     sanityCheckOngoingMovement();
```

Comment:
If the `Executor` thread context switch here, `shutdown()` won't know there's a need to `stopExecution()`.
When context switch back, this thread will go on to submit execution which would not be stopped.